src/dda/dda.c

```diff
@@ -0,0 +1,87 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   dda.c                                              :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: ycarro <ycarro@student.42.fr>              +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/02/22 14:48:17 by ycarro            #+#    #+#             */
+/*   Updated: 2023/02/23 17:42:22 by ycarro           ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "dda.h"
+
+void	dda(char **map, t_player *p, t_ray	*ray_data, t_point *collision)
+{
+	(void)map;
+	(void)collision;
+	set_incrementor(ray_data, ray_data->to_cast.radians);
+	get_ray_data(ray_data, ray_data->to_cast.radians, p);
+	while (!it_collides(ray_data, map, collision))
+	{
+		if (ray_differs_in_x(ray_data))
+		{
+			ray_data->x_pos.x += (double)(ray_data->incrementor.x * 100);
+			ray_data->x_pos.y = (ray_data->slope * ray_data->x_pos.x \
+				+ ray_data->intercept);
+		}
+		else
+		{
+			ray_data->y_pos.x += (double)(ray_data->incrementor.x * 100);
+			ray_data->y_pos.y = (ray_data->slope * ray_data->x_pos.x \
+				+ ray_data->intercept);
+		}
+	}
+}
+
+void	set_incrementor(t_ray *ray_data, double alpha)
+{
+	while (alpha > (2 * M_PI))
+		alpha -= (2 * M_PI);
+	if (alpha >= (3.0 / 2.0) * M_PI)
+	{
+		ray_data->incrementor.x = 1;
+		ray_data->incrementor.y = -1;
+	}
+	else if (alpha >= M_PI)
+	{
+		ray_data->incrementor.x = -1;
+		ray_data->incrementor.y = -1;
+	}
+	else if (alpha >= (1.0 / 2.0) * M_PI)
+	{
+		ray_data->incrementor.x = -1;
+		ray_data->incrementor.y = 1;
+	}
+	else
+	{
+		ray_data->incrementor.x = 1;
+		ray_data->incrementor.y = 1;
+	}
+}
+
+void	get_ray_data(t_ray *ray_data, double alpha, t_player *player)
+{
+	ray_data->slope = tan(alpha);
+	ray_data->intercept = player->pos.y - (player->pos.x * ray_data->slope);
+	ray_data->x_pos.x = player->pos.x;
+	ray_data->x_pos.y = player->pos.y;
+	ray_data->y_pos.x = player->pos.x;
+	ray_data->y_pos.x = player->pos.y;
+}
+
+int	ray_differs_in_x(t_ray *ray_data)
+{
+	if (ray_data->incrementor.x == 1)
```

Comment:
i think u can just muliplay by ray_data->incrementor.x
```
int tmp = ray_data->x_pos.x;
tmp *= ray_data->incrementor.x 
return (tmp >= (int)ray_data->y_pos.x)
```
im not sure but it looks pretty haha :)

Reply:
I dont think that should work. The incrementor indicates the direction we are moving, thats why it depend if its positive or negative. Imagine this case:
x = 100
y = 300
incrementor = 1
We should keep incrementing X till it is equal or bigger than Y
if (100 <=300) -> True
With just a mltiplication
return (100 >= 300) -> False